src/helpers/CatalogHelper.ts

```diff
@@ -0,0 +1,118 @@
+import { get } from 'lodash';
+
+export interface IIdentifierWithTags {
+    identifier: string;
+    tags: string;
+}
+
+export interface IDisplayForms {
+    [key: string]: IIdentifierWithTags;
+}
+
+export interface IAttrItem {
+    identifier: string;
+    tags: string;
+    displayForms: IDisplayForms;
+    defaultDisplayForm: IIdentifierWithTags;
+}
+
+export interface IAttrs {
+    [key: string]: IAttrItem;
+}
+
+export interface IDataSet {
+    identifier: string;
+    tags: string;
+    attributes: IAttrs;
+}
+
+export interface ICatalog {
+    metrics: { [key: string]: IIdentifierWithTags };
+    visualizations: { [key: string]: IIdentifierWithTags };
+    attributes: IAttrs;
+    dateDataSets: { [key: string]: IDataSet };
+}
+
+function getDisplayFormFromAttr(attrs: IAttrs, attributeName: string, displayFormName: string, subkey: string): string {
```

Comment:
[Q] What is `subkey`?

Reply:
Its a getter for the last "key" in `IIdentifierWithTags` - so it could be one of "identifier" or "tags". I can't think of better name, do you?

Reply:
`property`